README.md

```diff
@@ -1,62 +1,181 @@
+
+### Docker Compose
 
-In order to hack on specific pieces of the architecture you'll need to fetch those from Git.  Say to work on tooling invoker.
+We are using [Docker Compose](https://docs.docker.com/compose/).
+Compose is a tool for defining and running multi-container Docker applications.
+With Compose, you use a YAML file to configure your application’s services.
+Then, with a single command, you create and start all the services from your configuration.
+
+In the context of Exercism, each component (e.g. the website UI, an analyzer, the code that manages test runners) has it's own Dockerfile.
+These Dockerfiles are stored within the repositories (often named `dev.Dockerfile`) and are built and pushed to DockerHub via GitHub Actions.
+
+The code in this repository handles the creation of a `docker-compose.yml` and provides you with some simple wrapper scripts to run things.
+The `bin/start` script generates the `docker-compose.yml` from your `stack.yml` file, downloads the images from DockerHub and then starts them (via `docker-compose up`).
+
+### stack.yml
+
+The `stack.yml` file is a Exercism-specific configuration file that allows you to select which components you want to run locally, and any configuration you want to do.
+We then take that configuration and do the hard-work of turning that into a `docker-compose.yml`.
+
+Every docker image you download takes up HDD space, and every docker container you run takes up memory, so the `stack.yml` file allows to you ensure that you only run the things that are necessary for whichever part of Exercism you are working on.
+
+## Advanced Usage
+
+### Working on certain components of the architecture
+
+In order to working on a specific component of Exercism's architecture, you will need to have the relevant git repository downloaded locally.
+
+Presuming you don't change the directory name from the repo name on GitHub, running `./bin/start` will mount the locally checked out repository in the Docker container.
+This means any changes you make to the local filesystem are reflected within the Docker container.
+
+For example, if you wanted to work on the javascript-test-runner, you might do the following:
 
 ```bash
-# Again, this is the directory you're hosting all of Exercism repos within
+# Start in your general Exercism directory (the first you made in the instructions above)
 cd exercism
-git clone git@github.com:exercism/tooling-invoker.git
+
+# Clone the relevant repository
+git clone git@github.com:exercism/javascript-test-runner.git
+
+# Move into the directory for *this* repo
 cd v3-docker-compose
+
+#
+# Check the component is listed in the "enabled" section of your stack.yml.
+#
+
+# Run the bin/start command
+./bin/start
+```
+
+Depending on the reload-behaviour of the component you are working, you may need to restart the component after changes.
+For example, the Website live-updates any changes made to the code (because Rails), but the Tooling Invoker does not.
```

Comment:
```suggestion
For example, the Website live-updates any changes made to the code (because it is set-up and configured to do so), but the Tooling Invoker does not.
```

It's rails + webpacker + webpack + dev server + hmr or fast-refresh which are all separate technologies working together, but we probably don't want to say _that_.